src/skmultiflow/anomaly_detection/half_space_trees.py

```diff
@@ -0,0 +1,557 @@
+        self.anomaly_threshold = anomaly_threshold
+        self.is_learning_phase_on = True
+        self.tree_random_state = tree_random_state
+
+    def partial_fit(self, X, y, classes=None, sample_weight=None):
+        """ Partially (incrementally) fit the model.
+
+        Parameters
+        ----------
+        X : numpy.ndarray of shape (n_samples, n_features)
+            The features to train the model.
+
+        y: numpy.ndarray of shape (n_samples)
+            An array-like with the class labels of all samples in X.
+
+        classes: numpy.ndarray, optional (default=None)
+            Array with all possible/known class labels. This is an optional parameter, except
+            for the first partial_fit call where it is compulsory.
+
+        sample_weight: numpy.ndarray of shape (n_samples), optional (default=None)
+            Samples weight. If not provided, uniform weights are assumed.
+
+        Returns
+        -------
+        self
+
+        """
+
+        row_cnt, _ = X.shape
+
+        if self.samples_seen == 0:
+            self.build_trees()
+
+        for i in range(row_cnt):
+            self._partial_fit(X[i], y[i])
+
+        return self
+
+    def _partial_fit(self, X, y):
+        """ Trains the model on samples X and corresponding targets y.
+
+        Private function where actual training is carried on.
+
+        Parameters
+        ----------
+        X: numpy.ndarray of shape (1, n_features)
+            Instance attributes.
+
+        y: int
+            Class label for sample X.
+
+        """
+
+        # Populates first reference mass profile for every tree defined in the ensemble.
+        if self.samples_seen < self.window_size:
+            self.update_mass(X, True)
+        # Otherwise populates latest mass profile for every tree defined in the ensemble.
+        else:
+            self.update_mass(X, False)
+
+        if self.is_learning_phase_on is True and self.samples_seen > self.window_size:
+            self.set_is_learning_phase_on(False)
+
+        if (self.samples_seen % self.window_size) == 0:
+            self.update_models()
+
+        self.samples_seen += 1
+
+    def predict(self, X):
+        """ Predict classes for the passed data.
+
+        Parameters
+        ----------
+        X : numpy.ndarray of shape (n_samples, n_features)
+            The set of data samples to predict the class labels for.
+
+        Returns
+        -------
+        A numpy.ndarray with all the predictions for the samples in X.
+
+        """
+        r, _ = get_dimensions(X)
+        predictions = []
+        for i in range(r):
+            votes = self.get_votes_for_instance(X[i])
+            if votes == {}:
+                # Ensemble is empty, all classes equal, default to zero
+                predictions.append(0)
+            else:
+                predictions.append(max(votes, key=votes.get))
+        return np.asarray(predictions)
+
+    def get_votes_for_instance(self, X):
```

Comment:
This name is confusing, given that the method does not only retrieves the code but also determines the actual prediction. In this sense, I would prefer to rename it as `_predict` given that is the private function, called by `predict`, that actually determines the prediction value.

```python
   def _predict(self, X):
```